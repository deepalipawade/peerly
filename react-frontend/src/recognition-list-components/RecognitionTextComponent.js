import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Row } from "core-components/grid/GridComponent";
import { Form } from "core-components/form/FormComponent";
import ImageComponent from "core-components/image/ImageComponent";

const Border = styled.fieldset`
  border: 1px solid;
  border-radius: 11px;
  opacity: 1;
  margin-left: 5%;
  margin-right: 5%;
`;

const Legend = styled.legend`
  width: auto;
  margin-left: 10%;
`;

const Img = styled.div`
  width: 7vh;
  margin-right: 5%;
`;

const RecognitionTextComponent = ({ text, given_by }) => (
  <Row className="d-flex flex-column">
    <Border>
      <Legend>â€œ</Legend>
      <Form.Label className="px-2 pb-2 pt-0">{text}</Form.Label>
    </Border>
    <Row className="d-flex justify-content-end">
      <Form.Label className="font-weight-bold text-dark">{given_by}</Form.Label>
      <Img>
        <ImageComponent
          src="https://i.picsum.photos/id/2/200/200.jpg"
          roundedCircle={true}
          alt="Profile"
        />
      </Img>
    </Row>
  </Row>
);

RecognitionTextComponent.propTypes = {
  text: PropTypes.string.isRequired,
  given_by: PropTypes.string.isRequired,
};

export default React.memo(RecognitionTextComponent);
