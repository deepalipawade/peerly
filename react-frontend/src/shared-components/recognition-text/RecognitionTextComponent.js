import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import LabelComponent from "core-components/label/LabelComponent";

const Border = styled.div`
  background: #ffffff 0% 0% no-repeat padding-box;
  border: 1px solid #dae2e3;
  border-radius: 11px;
  opacity: 1;
`;

const TextAreaComponent = ({ text, className }) => (
  <Border>
    <LabelComponent className={className} text={text} />
  </Border>
);

TextAreaComponent.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
};

export default TextAreaComponent;
