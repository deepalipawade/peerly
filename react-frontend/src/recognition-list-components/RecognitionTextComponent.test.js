import React from "react";
import { render } from "@testing-library/react";

import RecognitionTextComponent from "recognition-list-components/RecognitionTextComponent";

describe("Recognition text Component component test", () => {
  test("should equal to recognition name", () => {
    const { getByText } = render(
      <RecognitionTextComponent
        text="give high five for good work"
        given_by="avinash"
      />
    );
    expect(getByText("avinash")).toHaveTextContent("avinash");
  });

  test("should equal to recognition text", () => {
    const { getByText } = render(
      <RecognitionTextComponent
        text="give high five for good work"
        given_by="avinash"
      />
    );
    expect(getByText("give high five for good work")).toHaveTextContent(
      "give high five for good work"
    );
  });

  test("render image component", () => {
    const { getByAltText } = render(
      <RecognitionTextComponent
        text="give high five for good work"
        given_by="avinash"
      />
    );
    expect(getByAltText("Profile")).toBeInTheDocument();
  });
});
