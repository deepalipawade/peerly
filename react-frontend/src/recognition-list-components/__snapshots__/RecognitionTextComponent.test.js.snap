// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition text Component test should equal snapshot 1`] = `
<DocumentFragment>
  <div>
    <fieldset
      class="sc-AxjAm cQYgKB"
    >
      <legend
        class="sc-AxirZ htcJuF"
      >
        â€œ
      </legend>
      <label
        class="px-2 pb-2 pt-0 form-label"
      >
        give high five
      </label>
    </fieldset>
    <div
      class="d-flex justify-content-end w-100"
    >
      <label
        class="font-weight-bold text-dark form-label"
      >
        Roy
      </label>
      <img
        alt="Profile"
        class="sc-AxiKw ilpYXo img-fluid rounded-circle"
        src="https://i.picsum.photos/id/2/200/200.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
